src/nccl_ofi_rdma.c

```diff
@@ -463,12 +463,47 @@ static inline int ofi_info_list_length(struct fi_info *info_list)
 	return length;
 }
 
+static inline int get_inject_rma_size_opt(nccl_net_ofi_device_t *base_dev,
```

Comment:
Why is this code specific to the rdma transport?  Why isn't it just part of the get_properties() call in nccl_ofi_net.c?

Reply:
Multiple reasons
1. It sets the `max_writelinine_size` for `iwriteInline()` function API function which is available in case of RMA support. So report RMA support only for RDMA protocol, and thus, only need to set `max_writeinline_size` in RDMA protocol case. 
2. When you look at the implementation, you see that it contains RDMA protocol specific pieces, like `nccl_net_ofi_rdma_ep_t`, `(&ep->rails[0].ofi_ep->fid`, ...

I think your question is targeting something specific?

Reply:
Yeah, these are all just pulling things from an endpoint, there's no reason that this should be rdma-specific code.

Reply:
You mean that I should open an of ep in net.c without the respective protocol? 

Or do you mean that each protocol should provide a new functionfunction to retrieve ofi ep? New, because there’s no protocol-agnostic way to go from a nccl_net_ofi_ep_t to a fi_ep, since RDMA has the rails and all that.

Reply:
or write a function that takes the fid_ep as the argument?

Reply:
You mean the `get_properties()` functions in sendrecv/RDMA retrieve an fid_ep, and call new function
```
static inline int get_inject_rma_size_opt(fid_ep ep, size_t *max_write_inline_size)
```
implemented in `net.c`?

Reply:
yes